### ISSUE TITLE
java.lang.Error: Defect in zio.Has on scala 3 if a trait is parametrized

### ISSUE THREAD

**Issue by leszekgruchala** [zio/izumi-reflect#225]
I have migrated my project from Scala 2.13.6 to 3.0.2 and it compiles using ZIO 1.0.12, but I have a zio-test spec that is failing on runtime with 
```
[info]     Fiber failed.
[info]     An unchecked error was produced.
[info]     java.lang.Error: Defect in zio.Has: Could not find RedisStream[=_$S|<Nothing..StreamInstance>] inside Map(package$::Sized$::Service -> zio.test.package$$anon$2@602d06c3,
```
The trait is defined as `trait RedisStream[S <: StreamInstance]` and I have a manually written mock for it using 
`object NotificationsRedisStreamMock extends Mock[Has[RedisStream[StreamInstance.Notifications]]]`

With Scala 2.13.6 everything is running well.

You can check out the code related to this from https://github.com/kensuio-oss/redis-streams-zio/tree/feature/scala-3
(branch `feature/scala-3`)

The trait definition: https://github.com/kensuio-oss/redis-streams-zio/blob/feature/scala-3/src/main/scala/io/kensu/redis_streams_zio/redis/streams/RedisStream.scala#L40
The mock definition: https://github.com/kensuio-oss/redis-streams-zio/blob/feature/scala-3/src/test/scala/io/kensu/redis_streams_zio/specs/mocks/NotificationsRedisStreamMock.scala
The failing test: https://github.com/kensuio-oss/redis-streams-zio/blob/feature/scala-3/src/test/scala/io/kensu/redis_streams_zio/redis/RedisConsumerSpec.scala

`sbt testOnly *.RedisConsumerSpec`

Potentially related to https://github.com/zio/izumi-reflect/issues/208 ?

Thank you for your effort!

**Comment by leszekgruchala** [comment]
Possible already fixed by https://github.com/zio/izumi-reflect/pull/198

Also related to https://github.com/zio/zio/issues/5530

**Comment by leszekgruchala** [comment]
I was able to overcome this problem thanks to this comment https://github.com/zio/zio/issues/4809#issuecomment-846545180

**Comment by kitlangton** [comment]
I believe this is now fixed due to #239

**Comment by leszekgruchala** [comment]
Closing then!
